### PR TITLE
Dependency version override prototype

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: sdk/parents/azure-sdk-parent/pom.xml [hunk 1]
```diff
@@ -43,6 +43,20 @@
...
+      <name>Spring Milestone Repository</name>
+      <url>https://repo.spring.io/milestone</url>
+
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
```
Comment (alzimmermsft) [inline sdk/parents/azure-sdk-parent/pom.xml:58]: I'd want to put this repository behind a profile if we can, I don't think it'd be best to use Spring Milestone as a default repository.

File: eng/versioning/update_versions.py [hunk 2]
```diff
@@ -229,6 +246,10 @@ def update_versions_all(update_type, build_type, target_file, skip_readme, auto_
         print('external_dependency_file=' + dependency_file)
         load_version_map_from_file(dependency_file, ext_dep_map)
 
+    if version_overrides:
+        load_version_overrides("eng/versioning/alternative_versions.yml", version_map, version_overrides)
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply eng/versioning/update_versions.py:250]: we don't need it right now, but eventually - why not? TestFromSource is some variation of it and perf testing does something similar. 

I'm happy to remove this now but want to have an approach that can work for our libs too.

**Pull request closed without merging**